handlers/messagebird/messagebird.go

```diff
@@ -0,0 +1,235 @@
+package messagebird
+
+/*
+ * Handler for MessageBird
+ */
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+
+	"encoding/json"
+
+	"fmt"
+
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+)
+
+var (
+	smsURL                    = "https://rest.messagebird.com/messages"
+	mmsURL                    = "https://rest.messagebird.com/mms"
+	signatureHeader           = "Messagebird-Signature-Jwt"
+	maxRequestBodyBytes int64 = 1024 * 1024
+)
+
+func init() {
+	courier.RegisterHandler(newHandler("MBD", "Messagebird", true))
+}
+
+type handler struct {
+	handlers.BaseHandler
+	validateSignatures bool
+}
+
+func newHandler(channelType courier.ChannelType, name string, validateSignatures bool) courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("MBD"), "Messagebird"), validateSignatures}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	s.AddHandlerRoute(h, http.MethodPost, "receive", courier.ChannelLogTypeMsgReceive, handlers.JSONPayload(h, h.receiveMessage))
+	return nil
+}
+
+func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request, payload *ReceivedMessage, clog *courier.ChannelLog) ([]courier.Event, error) {
+	err := h.validateSignature(channel, r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	// no message? ignore this
+	if payload.Body == "" && !payload.MMS {
+		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, channel, w, r, "Ignoring request, no message")
+	}
+
+	// create our date from the timestamp
+	date := payload.CreatedDatetime
+
+	// create our URN
+	urn, err := urns.NewTelURNForCountry(payload.Originator, "US")
```

Comment:
Why is the country here US only?
I think we want to use the channel country set in the DB